### ISSUE TITLE
[1.21.1] Unsafe `ItemProperty` registration

### ISSUE THREAD

**Issue by XFactHD** [kwpugh/Gobber_mod_forge#21]
Gobber registers its `ItemProperty`s directly in `FMLClientSetupEvent`. This is unsafe due to the client setup event firing in parallel and the backing maps used by `ItemProperties` not being thread-safe. Instead the item properties need to be registered in a deferred task enqueued via `FMLClientSetupEvent#enqueueWork()`.
This issue very likely also applies to older MC versions on MCForge as this code hasn't changed on the Minecraft/(Neo)Forge side in a very long time.

### Versions:
- NeoForge: 21.1.65
- Gobber: 1.21.1-2.10.15

### Stacktrace:
```
[Worker-Main-11/ERROR] [net.neoforged.fml.ModContainer/LOADING]: Caught exception during event FMLClientSetupEvent dispatch for modid gobber2
java.util.ConcurrentModificationException: null
	at java.util.HashMap.computeIfAbsent(HashMap.java:1229)
	at net.minecraft.client.renderer.item.ItemProperties.register(ItemProperties.java:61)
	at com.kwpugh.gobber2.util.BowUtils.setupBowModelProperties(BowUtils.java:41)
	at com.kwpugh.gobber2.events.BowModelEventHandler.setModelProperties(BowModelEventHandler.java:17)
	at net.neoforged.bus.EventBus.post(EventBus.java:350)
	at net.neoforged.bus.EventBus.post(EventBus.java:315)
	at net.neoforged.fml.ModContainer.acceptEvent(ModContainer.java:158)
	at net.neoforged.fml.ModLoader.lambda$dispatchParallelEvent$4(ModLoader.java:172)
	at net.neoforged.fml.ModLoader.lambda$dispatchParallelTask$7(ModLoader.java:231)
```
